passe à django 3

artragis opened the pull request:
### Contrôle qualité

Aller sur le site et valider que tout est ok. notamment la navigation entre les liens internes.

artragis committed:
passe à django 3

Situphen commented:
Il y a quelques points à investiguer avant de valider le passage à Django 3.1 https://github.com/zestedesavoir/zds-site/issues/6008

artragis committed:
django 3.1 bcbreak: ModelChoiceIteratorValue was introduced

Le plus gros effet que ça a sur nous c'est que du coup notre formulaire qui liste des catégories et qui habituellement recevait des string/int reçoit désormais cet objet.
Il faut donc déréférencer la value pour avoir l'objet désiré

artragis committed:
django 3.2 bcbreak: issue with drf

artragis committed:
back to 3.11 branch because of drf-extensions

Situphen reviewed (changes requested):
Je pense qu'il serait bien de fusionner cette PR *après* la v30.1 car ce n'est pas urgent et comme ça on aura plusieurs mois pour détecter de potentielles régressions. D'ailleurs amha l'idéal serait d'attendre que `drf-extensions` soit 3.2-compatible avant de fusionner pour être sur une LTS. (Je met "Request changes" pour bloquer la fusion.)

Situphen commented:
Remplacée par #6217
